firmware/thruna/DIM/src/app/states/app_driveState.c

```diff
@@ -110,7 +124,40 @@ static void driveStateRunOnTick100Hz(struct StateMachine *const state_machine)
     }
     else
     {
-        app_sevenSegDisplays_setGroup(SEVEN_SEG_GROUP_L, speed_kph);
+        uint32_t time_difference = current_time - previous_time;
+
+        if (globals->fault_in_car)
+        {
+            app_sevenSegDisplays_setGroup(SEVEN_SEG_GROUP_L, (float)fault_array[element_num]);
+            globals->element_num++;
+        }
+
+        else if (time_difference > 2000 && time_difference < 2500)
+        {
+            if (fault_num > 0)
+            {
+                app_sevenSegDisplays_setGroup(SEVEN_SEG_GROUP_L, (float)fault_array[element_num]);
+                globals->element_num  = 0;
+                globals->fault_in_car = true;
+            }
+
+            else if (element_num < warning_num)
+            {
+                app_sevenSegDisplays_setGroup(SEVEN_SEG_GROUP_L, (float)fault_array[element_num]);
+            }
+
+            else
+            {
+                app_sevenSegDisplays_setGroup(SEVEN_SEG_GROUP_L, speed_kph);
+                globals->element_num = 0;
+            }
+            globals->previous_time = previous_time;
+        }
+
+        else
+        {
+            app_sevenSegDisplays_setGroup(SEVEN_SEG_GROUP_L, speed_kph);
+        }
```

Comment:
Do you actually use your `app_sevenSegDisplays_setFaultCode` here?

Reply:
I didnt because I wanted to test the code with the original function we have already had to see if there is any point in keeping the function I made